alternator/rjson.cc

```diff
@@ -113,6 +113,48 @@ void push_back(rjson::value& base_array, rjson::value&& item) {
 
 }
 
+bool single_value_comp::operator()(const rjson::value& r1, const rjson::value& r2) const {
+       auto r1_type = r1.GetType();
+       auto r2_type = r2.GetType();
+       switch (r1_type) {
+       case rjson::type::kNullType:
+           return r1_type < r2_type;
+       case rjson::type::kFalseType:
+           return r1_type < r2_type;
+       case rjson::type::kTrueType:
+           return r1_type < r2_type;
+       case rjson::type::kObjectType:
+           throw rjson::error("Object type comparison is not supported");
+       case rjson::type::kArrayType:
+           throw rjson::error("Array type comparison is not supported");
+       case rjson::type::kStringType: {
+           const size_t r1_len = r1.GetStringLength();
+           const size_t r2_len = r2.GetStringLength();
+           size_t len = std::min(r1_len, r2_len);
+           int result = std::strncmp(r1.GetString(), r2.GetString(), len);
+           return result < 0 || (result == 0 && r1_len < r2_len);
+       }
+       case rjson::type::kNumberType: {
+           if (r1_type != r2_type) {
+               throw rjson::error("All numbers in a set should have the same type");
+           }
+           if (r1.IsDouble()) {
```

Comment:
These if()s, inside a switch(), are so strange - don't all these IsDouble, IsInt, etc. correspond to something that could be used in the switch()

Reply:
This code is taken straight from rjson internal implementations of its visitors. Using anything else was either not possible or relying on some undocumented internals, I don't remember which one.